filelog.go

```diff
@@ -87,6 +91,50 @@ func (w *FileLogWriter) Close() {
 	<- w.completed
 }
 
+// Track write failures and prints to stderr when possible. If err is nil, we'll try to clear the failures
+func (w *FileLogWriter) handleWriteFailure(err error) {
+	// Try to note any previous failures
+	if w.writeFailures != 0 {
+		_, fprintfErr := fmt.Fprintf(os.Stderr, "FileLogWriter(%q): Dropped %d previous log messages\n", w.filename, w.writeFailures)
+		if fprintfErr != nil {
+			// If we can't print now, exit early and try later
+			if err != nil { w.writeFailures += 1 }
+			return
+		} else {
+			w.writeFailures = 0
+		}
+	}
+	// If we have a current failure, attempt to print it
+	if err != nil {
+		_, fprintfErr := fmt.Fprintf(os.Stderr, "FileLogWriter(%q): Write failed: %v\n", w.filename, err)
+		if fprintfErr != nil {
+			w.writeFailures += 1
+		}
```

Comment:
@MarkRoddy I'm not sure under which condition two `Dropped previous log message` notices would occur, so I wrote a test where writes to the log and to the `errorWriter` fail, and it seems to work as expected.